docs/notes/2.23.x.md

```diff
@@ -12,6 +12,7 @@ We offer [formal sponsorship tiers for companies](https://www.pantsbuild.org/spo
 
 ### Highlights
 
+- Support documenting macro constants using `MY_CONSTANT: Annotated[some_type, "my help text ..."] = some_value`.
```

Comment:
I've been thinking of this highlights section as references to the best parts of things also listed later in the exhaustive enumeration... so: 

1. Could you add a call out later too? Maybe in a "Plugin API changes" section or similar?
2. I'm not sure this quite meets the bar for a highlight, as it seems a bit niche?

Reply:
(Example, to make that a bit more concrete: https://github.com/pantsbuild/pants/blob/8cc398868cc9adbb4a1593a9db93cd63225bcf1b/docs/notes/2.22.x.md#highlights )

Reply:
Thanks. Perhaps it would also help reduce merge conflicts if fewer PRs add highlights at all.. with notes spread across various sections, I anticipate it would be pretty straight forward to closer to release pick out the highlights from all the changes presented.

Reply:
Yeah, agreed. I think most PRs don't add highlights currently.